File: AverageSalaryExcludingTheMinimumAndMaximumSalary.java
// https://leetcode.com/problems/average-salary-excluding-the-minimum-and-maximum-salary/description/
// You are given an array of unique integers salary where salary[i] is the salary of the ith employee.
// Return the average salary of employees excluding the minimum and maximum salary. Answers within 10-5 of the actual answer will be accepted.

class Solution {
    public double average(int[] salary) {
        int min = salary[0];
        int max = salary[0];
        double sum = 0;

        for (int i = 0; i < salary.length; i++) {
            if (salary[i] < min) min = salary[i];
            if (salary[i] > max) max = salary[i];
        }

        for (int i = 0; i < salary.length; i++) {
            if (salary[i] != min && salary[i] != max) {
                sum += Double.valueOf(salary[i]);
            }
        }

        return sum / (salary.length - 2);
    }
}

File: AverageSalaryExcludingTheMinimumAndMaximumSalary.kt
// https://leetcode.com/problems/average-salary-excluding-the-minimum-and-maximum-salary/
// You are given an array of unique integers salary where salary[i] is the salary of the ith employee.
// Return the average salary of employees excluding the minimum and maximum salary. Answers within 10-5 of the actual answer will be accepted.

class Solution {
    fun average(salary: IntArray): Double {
        var max = salary[0]
        var min = salary[0]
        var sum = 0.0
        
        salary.forEach {
            if (it > max) max = it
            if (it < min) min = it
        }
        
        val salaryWithoutMaxAndMin = salary.filter { it != max && it != min }
        
        salaryWithoutMaxAndMin.forEach {
            sum += it.toDouble()
        }
        
        return (sum / salaryWithoutMaxAndMin.size.toDouble())
    }
}
